includes/widgets/table.php

```diff
@@ -0,0 +1,288 @@
+				'label_off' => esc_html__( 'Hide', 'elementor' ),
+				'return_value' => 'yes',
+				'default' => 'yes',
+			]
+		);
+
+		$this->add_control(
+			'column_one',
+			[
+				'label' => esc_html__( 'Title column 1', 'elementor' ),
+				'type' => Controls_Manager::TEXT,
+				'label_block' => true,
+				'placeholder' => esc_html__( 'Enter column title', 'elementor' ),
+				'dynamic' => [
+					'active' => true,
+				],
+			]
+		);
+
+		$this->add_control(
+			'column_two',
+			[
+				'label' => esc_html__( 'Title column 2', 'elementor' ),
+				'type' => Controls_Manager::TEXT,
+				'label_block' => true,
+				'placeholder' => esc_html__( 'Enter column title', 'elementor' ),
+				'dynamic' => [
+					'active' => true,
+				],
+			]
+		);
+		$this->add_control(
+			'column_three',
+			[
+				'label' => esc_html__( 'Title column 3', 'elementor' ),
+				'type' => Controls_Manager::TEXT,
+				'label_block' => true,
+				'placeholder' => esc_html__( 'Enter column title', 'elementor' ),
+				'dynamic' => [
+					'active' => true,
+				],
+			]
+		);
+
+		
+		$this->add_control(
+			'rows',
+			[
+				'label' => esc_html__( "Rows' contents", 'elementor' ),
+				'show_label' => esc_html__( true, 'elementor' ),
+				'type' => \Elementor\Controls_Manager::REPEATER,
+				'fields' => [
+					[
+						'name' => 'column_content_one',
+						'label' => esc_html__( 'Content column 1', 'elementor' ),
+						'type' => \Elementor\Controls_Manager::TEXT,
+						'default' => esc_html__( 'Content column 1' , 'elementor' ),
+						'label_block' => false,
+					],
+					[
+						'name' => 'column_content_two',
+						'label' => esc_html__( 'column content 2', 'elementor' ),
+						'type' => \Elementor\Controls_Manager::TEXT,
+						'default' => esc_html__( 'Content column 2' , 'elementor' ),
+						'label_block' => false,
+					],
+					[
+						'name' => 'column_content_three',
+						'label' => esc_html__( 'column content 3', 'elementor' ),
+						'type' => \Elementor\Controls_Manager::TEXT,
+						'default' => esc_html__( 'Content column 3' , 'elementor' ),
+						'label_block' => false,
+					],
+				],
+				'title_field' => esc_html__( 'row', 'elementor'),
+				
+			]
+			);
+
+		$this->end_controls_section();
+	}
+
+	/**
+	 * Render table widget output on the frontend.
+	 *
+	 * Written in PHP and used to generate the final HTML.
+	 *
+	 * @since 1.0.0
+	 * @access protected
+	 */
+	
+	protected function render() {
+		 $settings = $this->get_settings_for_display();
+
+		 echo '<table>';
+			if ( 'yes' === $settings['show_title'] ) {
+					echo '<thead>';
+						echo '<tr>';
+							echo '<th>' . $settings['column_one'] . '</th>';
+							echo '<th>' . $settings['column_two'] . '</th>';
+							echo '<th>' . $settings['column_three'] . '</th>';
+						echo '</tr>';
+					echo '</thead>';
+				}
+			
+			if ( $settings['rows'] ) {
+					echo '<tbody>';
+					$rows = $settings['rows'];
+					for ($i=0; $i < count($rows) - 1; $i++) { 
```

Comment:
I think the rows minus 1 is causing a problem.

Elementor used always spaces in between, check the other files in the repository.

```suggestion
					for ( $i=0; $i < count( $rows ); $i++ ) { 
```

Reply:
yes I should have put <= like in the JS side